### PR TITLE
Project information

### PR DESCRIPTION
**Pull request by MaximBazarov**

### CODE REVIEW

File: Docs/core.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "actions": [
+        {
+            "name": "SearchComplete",
+            "fields": [
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]: Move fields to another state leaf.

File: Docs/core.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    "type": "MoviesListProps"
+                },
+                {
+                    "name": "query",
+                    "type": "String"
+                }
+            ]
+        }
+    ],
+    "links": {
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]: Drop links section. All state should be 1 level nested.

File: Docs/core.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        "actions-to-states": [
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]: `-` is not a valid identifier. It will be simpler to not use it.

File: Docs/core.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            "name": "SearchState"
+        },
+        {
+            "name": "SuggestionsState"
+        }
+    ],
+    "props": [
+        {
+            "name": "SearchScreenProps",
+            "fields": [
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]: Move `fields` to separate state leaf.

File: Docs/core.json (hunk 3 above)
Comment (MaximBazarov) [reply Docs/core.json]: what do you think would be better? camel case or `_` instead?

File: Docs/core.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                "name": "results",
+                "type": "MoviesListProps"
+            },
+            {
+                "name": "query",
+                "type": "String"
+            }
+        ]
+    },
+    "actions2states": [
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]:
```
stateActions: {
  "State": [ "Action1", "Action2" ]
}
```

what do you think?

Comment (MaximBazarov) [reply Docs/core.json]: I think it's much better

File: Docs/core.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    "stateActions": {
+        "Search": [
+            "SearchComplete",
+            "SearchFailed"
+        ],
+        "Suggestions": [
+            "SearchComplete"
+        ]
+    },
+    "states2props": {
```
Comment (AlexeyDemedetskiy) [inline Docs/core.json]: stateProps?

**Comment by AlexeyDemedetskiy** [comment]
Should we also have `stateFields` for consistency?

**Comment by MaximBazarov** [comment]
added state fields as well

**Pull request merged**